src/languages/graphql.js

```diff
@@ -0,0 +1,75 @@
+/*
+ Language: graphql
+ Category: scripting, protocols, web
+ Author: John Foster [jf990][], and others
+ Description: Highlight GraphQL queries with highlight.js.
+*/
+/** @type LanguageFn */
+export default function(hljs) {
+  const regex = hljs.regex;
+  const GQL_NAME = /[_A-Za-z][_0-9A-Za-z]*/;
+  return {
+    name: "GraphQL",
+    aliases: [ "gql" ],
+    case_insensitive: true,
+    disableAutodetect: false,
+    keywords: {
+      keyword: [
+        "query",
+        "mutation",
+        "subscription",
+        "type",
+        "input",
+        "schema",
+        "directive",
+        "interface",
+        "union",
+        "scalar",
+        "fragment",
+        "enum",
+        "on"
+      ],
+      literal: [
+        "true",
+        "false",
+        "null"
+      ]
+    },
+    contains: [
+      hljs.HASH_COMMENT_MODE,
+      hljs.QUOTE_STRING_MODE,
+      hljs.NUMBER_MODE,
+      {
+        scope: "punctuation",
+        match: /[.]{3}/,
+        relevance: 0
+      },
+      {
+        scope: "punctuation",
+        begin: /[\!\(\)\:\=\[\]\{\|\}]{1}/,
```

Comment:
Should some of these not be `operator`?

Reply:
@joshgoebel I was trying to mimic the prism highlighting done on [graphql.org](https://graphql.org/learn/), if you inspect a sample on their doc pages the punctuation chars are given their own scope

```xml
<span class="token keyword">function</span>
<span class="token plain"> </span>
<span class="token function">Query_me</span>
<span class="token punctuation">(</span>
<span class="token plain">request</span>
<span class="token punctuation">)</span>
...
```

so I was trying to recreate that, at least initially, and it seems OK in my testing. I'm now leaning to leaving it in.

As for graphql operators, they are tagged as `and:` or `or:`, I did not apply special highlighting for those because they clash with regular properties and I was having difficulty getting them correct.

Reply:
Ok, we'll leave as-is for now and see how it goes.